airbyte-ci/connectors/metadata_service/orchestrator/tests/test_debug.py

```diff
@@ -41,25 +48,66 @@ def debug_registry_projection():
     # github_connector_folders_list = github_connector_folders(context).value
 
-    registry_derived_metadata_definitions(context, cloud_sources_df, cloud_destinations_df, oss_sources_df, oss_destinations_df).value
+    legacy_registry_derived_metadata_definitions(
+        context, cloud_sources_df, cloud_destinations_df, oss_sources_df, oss_destinations_df
+    ).value
     # valid_metadata_report_dataframe_df = valid_metadata_report_dataframe(metadata_definitions_df).value
 
     # all_sources_df = all_sources_dataframe(cloud_sources_df, oss_sources_df, github_connector_folders_list, valid_metadata_report_dataframe_df)
     # all_destinations_df = all_destinations_dataframe(cloud_destinations_df, oss_destinations_df)
 
     # connector_registry_location_html(context, all_sources_df, all_destinations_df)
     # connector_registry_location_markdown(context, all_sources_df, all_destinations_df)
+
+
+def debug_registry_generation():
+    resources = {
+        "gcp_gcs_client": gcp_gcs_client.configured(
+            {
+                "gcp_gcs_cred_string": {"env": "GCS_CREDENTIALS"},
+            }
+        ),
+        "gcs_bucket_manager": gcs_bucket_manager.configured({"gcs_bucket": {"env": "METADATA_BUCKET"}}),
+        "registry_directory_manager": gcs_file_manager.configured({"gcs_bucket": {"env": "METADATA_BUCKET"}, "prefix": REGISTRIES_FOLDER}),
+        "latest_metadata_file_blobs": gcs_directory_blobs.configured({"prefix": METADATA_FOLDER, "suffix": METADATA_FILE_NAME}),
+    }
+
+    context = build_op_context(resources=resources)
+    metadata_definitions_asset = metadata_definitions(context)
+    oss_registry_from_metadata(context, metadata_definitions_asset).value
+    # cloud_registry_from_metadata(context, metadata_definitions_asset).value
+
+
+def debug_registry_diff():
+    resources = {
+        "gcp_gcs_client": gcp_gcs_client.configured(
+            {
+                "gcp_gcs_cred_string": {"env": "GCS_CREDENTIALS"},
+            }
+        ),
+        "gcs_bucket_manager": gcs_bucket_manager.configured({"gcs_bucket": {"env": "METADATA_BUCKET"}}),
+        "registry_directory_manager": gcs_file_manager.configured({"gcs_bucket": {"env": "METADATA_BUCKET"}, "prefix": REGISTRIES_FOLDER}),
+        "latest_metadata_file_blobs": gcs_directory_blobs.configured({"prefix": METADATA_FOLDER, "suffix": METADATA_FILE_NAME}),
+        "legacy_oss_registry_gcs_blob": gcs_file_blob.configured({"prefix": "", "gcs_filename": "oss_catalog.json"}),
+    }
+
+    context = build_op_context(resources=resources)
+    metadata_definitions_asset = metadata_definitions(context)
+    oss_registry_from_metadata_value = oss_registry_from_metadata(context, metadata_definitions_asset).value
+    legacy_oss_registry_dict_value = legacy_oss_registry_dict(context)
+    legacy_oss_registry_value = legacy_oss_registry(legacy_oss_registry_dict_value)
+    oss_registry_diff(oss_registry_from_metadata_value, legacy_oss_registry_value)
```

Comment:
These functions are not called, do you want to keep them or transform them as tests?

Reply:
Keep them, again this file is currently our best way as devs to setup break points. 

Will convert them into tests after we have the diff lining up correctly :)